src/applicationManager.ts

```diff
@@ -528,3 +546,239 @@ async function getGraphAPIRoles(token: string, graphApiPermissions: string[]) {
+    `https://graph.microsoft.com/v1.0/applications/${applicationId}`,
+    {
+      method: "PATCH",
+      headers: {
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(body),
+    },
+  );
+  await errorHandler("updating application config", result);
+}
+
+/**
+ * Necessary where an app role is deleted, they must all be disabled first. Easier to disable by default and enable when updating.
+ */
+async function disableAppRoles(
+  applicationId: string,
+  token: string,
+  appRoles: AppRoles,
+) {
+  const app = await readApplication({ token, applicationId });
+  let appRolesJson: Array<AppRole> = app.appRoles;
+
+  // Finds pairs of displayName and ID of app roles to compare existing app roles with incoming app roles
+  const incomingAppRolePairs = appRoles.map(
+    ({ displayName, id }) => `name:${displayName}_id:${id}`,
+  );
+  const existingAppRolePairs = appRolesJson.map(
+    ({ displayName, id }) => `name:${displayName}_id:${id}`,
+  );
+  const commonPairs = existingAppRolePairs.filter((pair) =>
+    incomingAppRolePairs.includes(pair),
+  );
+
+  // Break here if app roles are the same as current file - not disabling all roles if unnecessary
+  if (!(commonPairs.length < existingAppRolePairs.length)) {
+    return;
+  }
+
+  // Keep fetched array of AppRoles but change enabled to false
+  const disabledAppRolesJson = appRolesJson.map(
+    (role) =>
+      ({
+        ...role,
+        isEnabled: false,
+      }) as AppRole,
+  );
+  await updateApplicationAppRoles({
+    token,
+    applicationId,
+    appRoles: disabledAppRolesJson,
+  });
+  console.log("Temporarily disabled app roles to allow updates");
+}
+
+/**
+ * Checks at least one of the group app role assignment mappings exists for a given entra group ID and app role ID, for a given application
+ */
+async function checkIfGroupAppRoleAssignmentExists({
+  token,
+  groupId,
+  appRoleId,
+  applicationId,
+}: {
+  token: string;
+  groupId: string;
+  appRoleId: string;
+  applicationId: string;
+}): Promise<boolean> {
+  const response = await fetch(
+    `https://graph.microsoft.com/v1.0/groups/${groupId}/appRoleAssignments`,
+    {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${token}`,
+        "Content-Type": "application/json",
+      },
+    },
+  );
+  const assignments = await response.json();
+  const assignmentExists = assignments.value.some(
+    (assignment: { appRoleId: string; resourceId: string }) =>
+      assignment.appRoleId === appRoleId &&
+      assignment.resourceId === applicationId,
+  );
+  return assignmentExists;
+}
+
+/**
+ * Assigns groups to App Roles for a given application
+ * @param groupId Azure entra group ID to assign an app role to
+ * @param appRoleId ID of the app role to assign the group to
+ * @param applicationId ID of the application to assign the group to app roles for
+ */
+async function updateApplicationGroupAssignments({
```

Comment:
`POST` method means creating a new resource, hence `createApplicationxxxx` would make more sense

Reply:
Good spot thanks - updated in https://github.com/hmcts/azure-app-proxy-manager/pull/122/commits/da506952d45ce0409ea2024c97c3a945f819c2de